### PR TITLE
LMS-367-[BE][Seeder & API] Class List

### PR DESCRIPTION
**Pull request by FranzDelosSantos**
## Issue Link
https://framgiaph.backlog.com/board/LMS?selectedIssueKey=LMS-367&assignee=417519

## Defintion of Done

- [x] it must have an API to fetch the classes /api/classes/<:company_id>
- [x] use serializer function to implement the logics, to prevent long codes in views
- [x] create a seeder that will generate 20 classes
- [x] each class must have at least 10 trainees (okay for now. same students to all class)
- [x] each class must have one trainor (okay for now. same trainor to all class)

## Steps to reproduce
```
docker exec -it lms-django bash
pipenv shell
pipenv run migrate
pipenv run seeder
pipenv run user-seeder

```

## Affected Components / Functionalities / Page
user.json
serializer.py
urls.py
views.py

## Test Cases
able to fetch 20 classes

## Notes
Created 200 trainee seeders

## Screenshots
pr11

### CODE REVIEW

File: client/src/shared/hooks/useUserSearch.ts
Comment (ragg09-sunasterisk) [inline client/src/shared/hooks/useUserSearch.ts:1]: what is this for? is this part of the task?

**Review by ragg09-sunasterisk — COMMENTED** [review]
The data should be dependent on the ID in the params, this current API fetch data regardless of what <:company_id> passed

class list

File: client/src/shared/hooks/useUserSearch.ts
Comment (FranzDelosSantos) [reply client/src/shared/hooks/useUserSearch.ts:1]: I will delete it ster

File: api/app_sph_lms/api/views.py [hunk 1]
```diff
@@ -215,5 +215,12 @@ def create(self, request, *args, **kwargs):
...
-    queryset = Class.objects.all()
     serializer_class = ClassSerializer
+    
+    def get_queryset(self):
+        company_id = self.kwargs.get('company_id')
+        queryset = Class.objects.filter(company_id=company_id)
+        if not queryset.exists():
+            queryset = Class.objects.none()
+        return queryset
+
```
Comment (ragg09-sunasterisk) [inline api/app_sph_lms/api/views.py:226]: no newline at the end of the line

**Review by ragg09-sunasterisk — APPROVED** [review]

**Review by Marielleregner7 — APPROVED** [review]
**APPROVED**
- [ ] created a seeder that generates 20 classes that has 10 trainees and a trainer.
- [ ] No ESLint error.

Comments: 5
Test time: 36 mins
Iterations: 3

Proof:
classlist
image

**Pull request merged**